### ISSUE TITLE
Error Message

### ISSUE THREAD

**Issue by Kashdeya** [UniversalLP/JustEnoughButtons#27]
I have just updated my pack and added in Your mod. But I have noticed that when I click on the Magnet Mode button I get the following message in chat.

I'm sorry Kashdeya, I'm afraid I can't let you do that. 

I am in SP Survival, Is there something I am missing? If you need a list of mods You can find the pack on Curse (The Relax Pack) 

There is no other problem I have found seeing that I only use the a couple of buttons, Delete, Magnet and Save.

**Comment by univrsal** [comment]
Took a quick look at it and my first question would be do you have cheats enabled?
It works fine for me here with cheats enabled but without JEB won't work (Since most of the buttons work over commands that require cheats) and send that chat message.
Technically it should send the "You don't have permissions to use that command" message but it seems like I forgot to change that.

**Comment by Kashdeya** [comment]
no i do not have cheats enabled. So what you are saying is I have to turn cheats on to use magnet? 
if that is the case then that is a bad way to let users on a server have magnet mode. IMO. Cause I do not want to have to turn cheats on a server so people can have magnet mode. cause then they can do whatever they want. 

Thank you for the info, ill look for another way to add magnet mode.

**Comment by univrsal** [comment]
Server? You said you were in Singleplayer. On a server you have to be opped to use the magnet mode.

**Comment by Kashdeya** [comment]
no what I am saying is that NEI does not need to have cheats turned on to use magnet mode, So why does your mod need to have cheats on to use it? That just does not make sense at all. So you want me to turn cheats on every time i load of a game to play just to use magnet mode? why it should be there for everyone to use not just the people who play with cheats on. 

Same goes for servers, You should not have to be OP or in cheat mode to use magnet mode, there should be a setting to allow all or OP only on servers and you should not have to be in cheat mode in SP worlds to use it. It is just crazy.

**Comment by univrsal** [comment]
> Since most of the buttons work over **commands that require cheats**

The way this mod works is over commands. Setting the time to day or clearing your inventory is nothing more than using the corresponding commands. Therefore cheats have to be enabled for them to work.
This applies to almost every feature excluding inventory saves and the magnet mode (when JEB is installed on the server or used in single player).
The reason I did this was to allow this mod to work solely on the client side.

Magnet mode and Inventory saves could work without cheats enabled but since all the other functions need cheats enabled it made the most sense to only allow them to be used when cheats are on.
After all those functions are cheats and shouldn't be handed out to an entire server without permission checking.

TL;DR
I can add an option to allow magnet mode without OP in MP and allow the use of all functions in SP without cheats.

**Comment by Kashdeya** [comment]
All i'm saying is I'm a pack maker and a mod dev. I understand why you did what you did as a mod dev, but as a pack maker it just does not fit into any pack that I can think of, to ask the player to turn cheats on every time they make a new world just to use magnet mode. Cause a lot of people love magnet mode and love the trash can so in my packs those are the only two things I ever turn on for everyone. 

This way they get to choose to use them or not, but they way you have it set up is if people want to use your buttons in single player they have to have cheats on and a lot of player don't like to do that. I understand having to turn cheats on for weather, time of day and stuff but magnet mode should not be hindered behind cheat mode IMO. 

now as for server well there should be a config for it that allows the server admin to choice between OP, ALL or NONE. for all of your buttons, this would open up more options for the server admins and the mods of that server. 

I have learned the more flexible and configurable your mod is the more people love it. But this is all IMO I am just stating what I see wrong as a modpack dev. and why i rather use NEI as a add on for JEI then JEB. 

Thank you for your time, It is a great mod no the less. A lot of people needed some thing like this. It just is not for me.
